fili-core/src/test/groovy/com/yahoo/bard/webservice/table/resolver/DataSourceConstraintSpec.groovy

```diff
@@ -0,0 +1,33 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.table.resolver
+
+import com.yahoo.bard.webservice.data.metric.MetricColumn
+import spock.lang.Specification
+
+class DataSourceConstraintSpec extends Specification {
+
+    def "test withIntersectedColumnNames"() {
+        given:
```

Comment:
Use @Unroll and use #arguments in test definition to explain the parameters of each test case:
e.g.
#metricNames intersected with #other produces #newMetricNames